### PR TITLE
Add custom dimensions

### PR DESCRIPTION
**Pull request by kylerush**
This PR adds a plan type custom dimension in GA. This will be used for reporting and to tailor our remarketing ad lists.

You should see a "cd1" URL attribute on requests to `google-analytics.com` (custom events). The value will be either "free", "bronze", "silver", "gold", "enterprise", or "none" based on what your plan is. To reproduce this, run your local server, sign in, then look at the HTTP requests to `google-analytics.com`. You should see a "cd1" property on these requests with the correct value.

I added tests for this behavior. To make the test fail:

1. Change the `plan_id` value in website-guts/endpoint-mocks/ to any other plan, like `enterprise-monthly`, `enterprise-oneyear`, `enterprise-twoyear`, `gold-oneyear`, etc.
2. Restart your server
3. Run `grunt ui-test:misc`
4. See the test fail

### CODE REVIEW

**Comment by kylerush** [comment]
@dtothefp can you review?

File: website-guts/templates/layouts/wrapper.hbs [hunk 1]
```diff
@@ -27,8 +27,43 @@ layout_modals:
...
+          			'silver-twoyear': 		'3',
+          			'gold-monthly': 			'4',
+          			'gold-oneyear': 			'4',
+          			'gold-twoyear': 			'4',
+          			'enterprise-monthly': '5',
+          			'enterprise-oneyear': '5',
+          			'enterprise-twoyear': '5',
+                '1':                  'free',
+                '2':                  'bronze',
+                '3':                  'silver',
```
Comment (stewsmith) [inline website-guts/templates/layouts/wrapper.hbs]: This spacing is wack. Is this atom's doing?

**Comment by stewsmith** [comment]
@kylerush What is the status of this?

File: website-guts/templates/layouts/wrapper.hbs (hunk 1 above)
Comment (kylerush) [inline website-guts/templates/layouts/wrapper.hbs]: @stewsmith i don't think so. it just shows up this way in github for some reason. i don't know why this happens.

**Comment by kylerush** [comment]
@stewsmith can you review?

File: website-guts/templates/layouts/wrapper.hbs [hunk 2]
```diff
@@ -27,8 +27,49 @@ layout_modals:
...
+            window.optly = {}; window.analytics=window.analytics||[],window.analytics.methods=["identify","group","track","page","pageview","alias","ready","on","once","off","trackLink","trackForm","trackClick","trackSubmit"],window.analytics.factory=function(t){return function(){var a=Array.prototype.slice.call(arguments);return a.unshift(t),window.analytics.push(a),window.analytics}};for(var i=0;i<window.analytics.methods.length;i++){var key=window.analytics.methods[i];window.analytics[key]=window.analytics.factory(key)}window.analytics.load=function(t){if(!document.getElementById("analytics-js")){var a=document.createElement("script");a.type="text/javascript",a.id="analytics-js",a.async=!0,a.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.io/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(a,n)}},window.analytics.SNIPPET_VERSION="2.0.9",
+            window.optly.identifyVisitorAttributes = function(){
+              window.optly.planMap = {
```
Comment (stewsmith) [inline website-guts/templates/layouts/wrapper.hbs]: I would welcome a comment here. What do the number values mean?

File: website-guts/templates/layouts/wrapper.hbs [hunk 3]
```diff
@@ -27,8 +27,49 @@ layout_modals:
...
+                window.optly.visitorAttributes = window.optly.visitorAttributes.split('|');
+                if(typeof window.optly.planMap[window.optly.visitorAttributes[0]] === 'string'){
+                  plan = window.optly.planMap[window.optly.visitorAttributes[0]];
+                } else {
+                  plan = 'none';
+                }
+              } else {
+                plan = 'none';
+              }
+              window.console.log('setting plan to: ' + plan);
```
Comment (stewsmith) [inline website-guts/templates/layouts/wrapper.hbs]: Do you want to leave this in?

File: test/misc/custom-dimension-spec.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+var Nightmare = require('nightmare');
+var config = require('../config')({dirname: __dirname});
+var phantomPath = config.phantomPath;
+var testPath = config.basePath({path: '/'});
+
+describe('testing the signin, create account, retrieve password dialogs', function() {
+
+  describe('filling out the signin form', function() {
+
+    it('signs the user in', function(done) {
```
Comment (stewsmith) [inline test/misc/custom-dimension-spec.js]: Could you name this more accurately? Looks like you're evaluating the plan type, not signing the user in.

File: website-guts/templates/layouts/wrapper.hbs (hunk 2 above)
Comment (kylerush) [inline website-guts/templates/layouts/wrapper.hbs]: @stewsmith fair point, but i think it's probably not a good idea to have a public description of our plan types/codes.

**Pull request closed without merging**